File: scripts/refactor_html.py
import os
import re

def refactor_html_files():
    pages_dir = 'pages'
    root_dir = '.'
    
    # 1. Refactor root 404.html
    root_404 = os.path.join(root_dir, '404.html')
    if os.path.exists(root_404):
        with open(root_404, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Replace script src
        content = content.replace('assets/js/tailwind-config.js', 'configs/tailwind-config.js')
        
        with open(root_404, 'w', encoding='utf-8') as f:
            f.write(content)
        print("Refactored root 404.html")

    # 2. Refactor pages/
    inline_script_pattern = re.compile(r'<script\s+id="tailwind-config">.*?</script>', re.DOTALL)
    
    for filename in os.listdir(pages_dir):
        if filename.endswith('.html'):
            filepath = os.path.join(pages_dir, filename)
            with open(filepath, 'r', encoding='utf-8') as f:
                content = f.read()
            
            original = content
            
            # Replace inline tailwind configs
            if inline_script_pattern.search(content):
                content = inline_script_pattern.sub('<script src="../configs/tailwind-config.js"></script>', content)
                print(f"Removed inline tailwind config in {filename}")
                
            # Replace regular scripts
            content = content.replace('../assets/js/tailwind-config.js', '../configs/tailwind-config.js')
            content = content.replace('assets/js/tailwind-config.js', '../configs/tailwind-config.js')
            
            if content != original:
                with open(filepath, 'w', encoding='utf-8') as f:
                    f.write(content)
                print(f"Updated {filename}")

if __name__ == '__main__':
    refactor_html_files()

File: scripts/refactor_html.js
const fs = require('fs');
const path = require('path');

function refactorHtmlFiles() {
    const pagesDir = 'pages';
    const rootDir = '.';
    
    // 1. Refactor root 404.html
    const root404 = path.join(rootDir, '404.html');
    if (fs.existsSync(root404)) {
        let content = fs.readFileSync(root404, 'utf8');
        content = content.replace(/assets\/js\/tailwind-config\.js/g, 'configs/tailwind-config.js');
        fs.writeFileSync(root404, content, 'utf8');
        console.log("Refactored root 404.html");
    }

    // 2. Refactor pages/
    const files = fs.readdirSync(pagesDir);
    const inlineScriptPattern = /<script\s+id="tailwind-config">[\s\S]*?<\/script>/g;
    
    files.forEach(filename => {
        if (filename.endsWith('.html')) {
            const filepath = path.join(pagesDir, filename);
            let content = fs.readFileSync(filepath, 'utf8');
            const original = content;
            
            // Replace inline tailwind configs
            if (inlineScriptPattern.test(content)) {
                content = content.replace(inlineScriptPattern, '<script src="../configs/tailwind-config.js"></script>');
                console.log(`Removed inline tailwind config in ${filename}`);
            }
            
            // Replace regular scripts
            content = content.split('../assets/js/tailwind-config.js').join('../configs/tailwind-config.js');
            content = content.split('assets/js/tailwind-config.js').join('../configs/tailwind-config.js');
            
            if (content !== original) {
                fs.writeFileSync(filepath, content, 'utf8');
                console.log(`Updated ${filename}`);
            }
        }
    });
}

refactorHtmlFiles();
